Makefile

```diff
@@ -79,7 +79,10 @@ cover:
 
 format:
 	@echo ">> formatting code"
-	GO111MODULE=$(GO111MODULE) $(GO) fmt $(pkgs)
+	# Replace gofmt call once we bump to a more recent Go that supports `-mod=vendor`, probably 1.14.
+	# GO111MODULE=$(GO111MODULE) $(GO) fmt $(GOOPTS) $(pkgs)
+	# Avoid formatting anything under vendor/.
+	$(GOFMT) -l -w $(shell find . -path ./vendor -prune -o -name '*.go' -print)
```

Comment:
I'm by no means a Go expert, and I realize that this is the same construct as what's used above, but would it be cleaner to use the approach from [this gist](https://gist.github.com/apremalal/b7cc4ec9047f0a5dd52a28bad0fbb708):
```sh
GO111MODULE=$(GO111MODULE) $(GO) fmt $(shell $(GO) list $(pkgs) | grep -v '/vendor/')
```
?

Reply:
The PR description explains why we can't use `go fmt`. Can you help me understand the benefits of the version you are suggesting?

Reply:
Sorry for being unclear — the suggestion was to use `go list` instead of `find`. I believe it's cleaner because it doesn't make any assumptions about the value of `$(pkgs)`, and will work with Go's recursive subdirectory paths (`...`). You could use it with `$(GOFMT)` as well.

Reply:
That command resolves into the following command-line, which doesn't work:

```sh
$ gofmt -l -w github.com/Stackdriver/stackdriver-prometheus-sidecar/bench github.com/Stackdriver/stackdriver-prometheus-sidecar/cmd/stackdriver-prometheus-sidecar github.com/Stackdriver/stackdriver-prometheus-sidecar/metadata github.com/Stackdriver/stackdriver-prometheus-sidecar/retrieval github.com/Stackdriver/stackdriver-prometheus-sidecar/stackdriver github.com/Stackdriver/stackdriver-prometheus-sidecar/tail github.com/Stackdriver/stackdriver-prometheus-sidecar/targets
stat github.com/Stackdriver/stackdriver-prometheus-sidecar/bench: no such file or directory
stat github.com/Stackdriver/stackdriver-prometheus-sidecar/cmd/stackdriver-prometheus-sidecar: no such fil
e or directory                                                                                           
stat github.com/Stackdriver/stackdriver-prometheus-sidecar/metadata: no such file or directory
stat github.com/Stackdriver/stackdriver-prometheus-sidecar/retrieval: no such file or directory
stat github.com/Stackdriver/stackdriver-prometheus-sidecar/stackdriver: no such file or directory
stat github.com/Stackdriver/stackdriver-prometheus-sidecar/tail: no such file or directory
stat github.com/Stackdriver/stackdriver-prometheus-sidecar/targets: no such file or directory
make: *** [Makefile:81: format] Error 2
```

Reply:
I see. I was confused. I thought this was simply about filtering out `vendor` (i.e., golang/go#19090), but "go fmt" already does that.
For posterity, the problem was that any `go` command (including the `go list` I suggested) will download the modules from GitHub, which is what you want to avoid, unless the `-mod=vendor` flag is specified. Since `go fmt` does not respect `-mod`, you need to supply the files to the underlying command explicitly.
I didn't have sufficient context to even understand the PR description until now.